bundles/org.openhab.core.io.rest.auth/src/main/java/org/openhab/core/io/rest/auth/internal/AuthFilter.java

```diff
@@ -25,33 +27,72 @@
 import org.openhab.core.auth.Authentication;
+import org.openhab.core.auth.User;
+import org.openhab.core.auth.UserRegistry;
+import org.openhab.core.auth.UsernamePasswordCredentials;
+import org.openhab.core.config.core.ConfigurableService;
 import org.openhab.core.io.rest.JSONResponse;
 import org.openhab.core.io.rest.RESTConstants;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
 import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.jaxrs.whiteboard.JaxrsWhiteboardConstants;
 import org.osgi.service.jaxrs.whiteboard.propertytypes.JaxrsApplicationSelect;
 import org.osgi.service.jaxrs.whiteboard.propertytypes.JaxrsExtension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * This filter is responsible for parsing a token provided with a request, and hydrating a {@link SecurityContext} from
- * the claims contained in the token.
+ * This filter is responsible for parsing credentials provided with a request, and hydrating a {@link SecurityContext}
+ * from these credentials if they are valid.
  *
  * @author Yannick Schaus - initial contribution
+ * @author Yannick Schaus - Allow basic authentication
  */
 @PreMatching
-@Component
+@Component(configurationPid = "org.openhab.restauth", property = Constants.SERVICE_PID + "=org.openhab.restauth")
+@ConfigurableService(category = "system", label = "API Security", description_uri = AuthFilter.CONFIG_URI)
 @JaxrsExtension
 @JaxrsApplicationSelect("(" + JaxrsWhiteboardConstants.JAX_RS_NAME + "=" + RESTConstants.JAX_RS_NAME + ")")
 @Priority(Priorities.AUTHENTICATION)
 @Provider
 public class AuthFilter implements ContainerRequestFilter {
+    private final Logger logger = LoggerFactory.getLogger(AuthFilter.class);
+
     private static final String ALT_AUTH_HEADER = "X-OPENHAB-TOKEN";
 
+    protected static final String CONFIG_URI = "system:restauth";
+    private static final String CONFIG_ALLOW_BASIC_AUTH = "allowBasicAuth";
+    private static final String CONFIG_NO_UNAUTHENTICATED_USER_ROLE = "noUnauthenticatedUserRole";
+
+    private boolean allowBasicAuth = false;
+    private boolean noUnauthenticatedUserRole = false;
+
     @Reference
     private JwtHelper jwtHelper;
 
+    @Reference
+    private UserRegistry userRegistry;
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    protected void modified(@Nullable Map<String, @Nullable Object> properties) {
+        if (properties != null) {
+            Object value = properties.get(CONFIG_ALLOW_BASIC_AUTH);
+            allowBasicAuth = value != null && "true".equals(value.toString());
+            value = properties.get(CONFIG_NO_UNAUTHENTICATED_USER_ROLE);
+            noUnauthenticatedUserRole = value != null && "true".equals(value.toString());
```

Comment:
Argh, I am really getting a knot in my brain here (German proverb) by this double negation. How about changing this parameter to something like "enableAuthForUserResources"?

Reply:
It was more to the tune of "prevent unauthenticated requests to get an implicit user role in the security context", so I tried to rename it to "noImplicitUserRole" / "No implicit user role for unauthenticated requests", that's perhaps clearer. Still a double negation, so if you still disagree I'll change it to "enableAuthForUserResources" :)

Reply:
And if we change it to "ImplicitUserRole" and have default=true, knowing that on the long run our wish is to get the setting to OFF? It would be in line with the basic auth setting, i.e. enabling something means your system becomes less secure.

Reply:
Good point, I insisted on having false as the default but that's not really a requirement.